packages/admin-next/dashboard/src/hooks/api/order-edits.tsx

```diff
@@ -0,0 +1,202 @@
+import { ordersQueryKeys } from "./orders"
+
+export const useCreateOrderEdit = (
+  orderId: string,
+  options?: UseMutationOptions<
+    HttpTypes.AdminOrderEditPreviewResponse,
+    Error,
+    HttpTypes.AdminInitiateOrderEditRequest
+  >
+) => {
+  return useMutation({
+    mutationFn: (payload: HttpTypes.AdminInitiateOrderEditRequest) =>
+      sdk.admin.orderEdit.initiateRequest(payload),
+    onSuccess: (data: any, variables: any, context: any) => {
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.details(),
+      })
+
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.preview(orderId),
+      })
+      options?.onSuccess?.(data, variables, context)
+    },
+    ...options,
+  })
+}
+
+export const useRequestOrderEdit = (
+  id: string,
+  options?: UseMutationOptions<
+    HttpTypes.AdminOrderEditPreviewResponse,
+    Error,
+    void
+  >
+) => {
+  return useMutation({
+    mutationFn: () => sdk.admin.orderEdit.request(id),
+    onSuccess: (data: any, variables: any, context: any) => {
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.details(),
+      })
+
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.preview(id),
+      })
+      options?.onSuccess?.(data, variables, context)
+    },
+    ...options,
+  })
+}
+
+export const useConfirmOrderEdit = (
+  id: string,
+  options?: UseMutationOptions<
+    HttpTypes.AdminOrderEditPreviewResponse,
+    Error,
+    void
+  >
+) => {
+  return useMutation({
+    mutationFn: () => sdk.admin.orderEdit.confirm(id),
+    onSuccess: (data: any, variables: any, context: any) => {
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.details(),
+      })
+
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.preview(id),
+      })
+      options?.onSuccess?.(data, variables, context)
+    },
+    ...options,
+  })
+}
+
+export const useCancelOrderEdit = (
+  orderId: string,
+  options?: UseMutationOptions<any, Error, any>
+) => {
+  return useMutation({
+    mutationFn: () => sdk.admin.orderEdit.cancelRequest(orderId),
+    onSuccess: (data: any, variables: any, context: any) => {
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.details(),
+      })
+
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.preview(orderId),
+      })
+      options?.onSuccess?.(data, variables, context)
+    },
+    ...options,
+  })
+}
+
+export const useAddOrderEditItems = (
+  id: string,
+  options?: UseMutationOptions<
+    HttpTypes.AdminOrderEditPreviewResponse,
+    Error,
+    HttpTypes.AdminAddOrderEditItems
+  >
+) => {
+  return useMutation({
+    mutationFn: (payload: HttpTypes.AdminAddOrderEditItems) =>
+      sdk.admin.orderEdit.addItems(id, payload),
+    onSuccess: (data: any, variables: any, context: any) => {
+      queryClient.invalidateQueries({
+        queryKey: ordersQueryKeys.preview(id),
+      })
```

Comment:
sugg: In some places we check for order as well. You can end up in a stale state when you exit the modal without performing an action

```suggestion
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.details(),
      })
      
      queryClient.invalidateQueries({
        queryKey: ordersQueryKeys.preview(id),
      })
```

Reply:
Hmm, have you encountered a flow with OE that resulted in such a state? It shouldn't be possible since all OE actions "operate" on preview and the order shouldn't be affected until the edit is confirmed.

When we exit the modal cancel hook is called which has details invalidation.

Reply:
Is this something we can have in mind for testing, so we can get this merged?

Reply:
Yes, I think it's good for merge. We can add additional invalidations if we see the need in a followup.

Reply:
Yes, none of these are blockers. Let's get this bad boy in